app/v2/models/Payload.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.models
+
+import play.api.libs.json.Format
+import play.api.libs.json.JsString
+import play.api.libs.json.Json
+
+object Payload {
+  implicit lazy val payloadFormat: Format[Payload] = Json.valueFormat[Payload]
+}
+
+case class Payload(value: String) {
+  lazy val asJson = Json.parse(value)
+  lazy val asXml  = JsString(value)
+}
```

Comment:
Ah, okay, this isn't quite what we meant yesterday, rather:

```scala
abstract class Payload(val value: String) {
    def asJson: JsValue
}

case class XMLPayload(value: String) extends Payload(value) {
    lazy val asJson: JsValue = JsString(value)
}

case class JsonPayload(value: String) extends Payload(value) {
    lazy val asJson: JsValue = Json.parse(value)
}
```

Then in the message summary reads, we'll need to specifically choose `XMLPayload` -- however, in the `MessageSummary` itself, the body must be of type `Payload`. For the cases we need to change from XML to Json, we swap the XMLPayload for the JsonPayload